main/src/ca/uwaterloo/flix/api/lsp/provider/ShowAstProvider.scala

```diff
@@ -32,20 +35,21 @@ object ShowAstProvider {
     * - `title` (a string like `Namer.flix.ir`)
     * - `text` (a string with the ir representation).
     */
-  def showAst(phase: String)(implicit index: Index, root: Option[Root], flix: Flix): JObject = root match {
-    case None =>
-      val text = "No IR available. Does the program not compile?"
-      astObject(phase, text)
-    case Some(r) =>
-      // We have to compile the program to obtain the relevant AST.
-      flix.codeGen(r)
+  def showAst(phase: String)(implicit index: Index, root: Option[Root], flix: Flix): JObject = {
+    val closest = closestPhaseName(phase)
+    val oldOpts = flix.options
+    flix.setOptions(oldOpts.copy(xprintphase = Set(closest)))
+    flix.compile()
+    flix.setOptions(oldOpts)
+    pathObject(AstPrinter.phaseOutputPath(closest))
+  }
 
-      val phasesWithNames = AstPrinter.allPhases().map { case (name, printer) => (name, (name, printer)) }
-      val (closestPhase, closestPrettyPrinter) = Similarity.closestMatch(phase, phasesWithNames)
-      astObject(closestPhase, closestPrettyPrinter())
+  private def closestPhaseName(phase: String) = {
+    val phaseNames = AstPrinter.phaseNames().m1.keys
+    Similarity.closestMatch(phase, phaseNames.map(s => (s, s)).toMap)
   }
 
-  private def astObject(phase: String, text: String): JObject = {
-    ("title" -> s"$phase.$IrFileExtension") ~ ("text" -> text)
+  private def pathObject(path: Path): JObject = {
+    ("path" -> path.toAbsolutePath.toString)
```

Comment:
Now i return an object with a path value. I'm not fully sure how to make this path into a string that the receiver will understand but here is something. To note for implementing the other side is that
- show ast always triggers a full recompile, that might fail
- the returned path *should* exist but it might not if we mess up in the list of phase names. it would be nice if that doesn't crash vscode.